rattr/config/_types.py

```diff
@@ -203,6 +203,29 @@ class Config(metaclass=ConfigMetaclass):
     PLUGINS_BLACKLIST_PATTERNS: set[str] = field(default_factory=set)
     """The set of blacklist patterns set by custom plugins."""
 
+    if TYPE_CHECKING:
+
+        @overload
+        def __init__(self) -> None:  # type: ignore[reportNoOverloadImplementation]
+            ...
+
+        @overload
+        def __init__(  # type: ignore[reportNoOverloadImplementation]
+            self,
+            arguments: Arguments,
+            state: State,
+        ) -> None:
+            ...
+
+        @overload
```

Comment:
Yes, unfortunately.

On the first call we optionally take `arguments` and `state`, to set the initial values. Subsequent calls to `Config()` returns the singleton. But the call `Config()` is seen as a type error by default because pylance etc has poor support for attrs

Reply:
Ah gotcha